examples/hello-world.mjs

```diff
@@ -1,16 +1,58 @@
+// Import necessary modules and initialize environment configuration
 import { SayariClient } from "../dist/index.js";
 
-// NOTE: To connect you most provide your client ID and client secret. To avoid accidentally checking these into git,
-// it is recommended to use ENV variables
+// Dynamically import `dotenv` and load environment variables only once
+const dotenv = await import("dotenv");
+dotenv.config();
+
+// Load and validate credentials from environment variables
 const clientID = process.env.CLIENT_ID;
 const clientSecret = process.env.CLIENT_SECRET;
 
-// Create an authenticated sayari client
-const client = new SayariClient({ clientId: clientID, clientSecret: clientSecret })
+// CLIENT_ID and CLIENT_SECRET - check if present
+if (!clientID || !clientSecret) {
+    console.error(
+        "Error:   CLIENT_ID and CLIENT_SECRET must be set in the environment variables \n \
+        Please refer to Sayari documentation!"
+    );
+    process.exit(1);
+}
+
+/**
+ * Initializes and returns an authenticated Sayari client
+ * @returns {SayariClient} - Authenticated Sayari client instance
+ */
+function createSayariClient() {
+    try {
+        return new SayariClient({ clientId: clientID, clientSecret: clientSecret });
+    } catch (error) {
+        console.error("Failed to initialize SayariClient:", error.message);
+        process.exit(1);
+    }
+}
+
+/**
+ * Resolves an entity and fetches its details
+ * @param {string} name - The name of the entity to resolve
+ */
+async function resolveEntityDetails(name) {
+    const client = createSayariClient();
+    try {
+        // Resolve entity
+        const resolution = await client.resolution.resolution({ name });
+        if (!resolution.data || resolution.data.length === 0) {
+            console.log(`No entity found for the name: ${name}`);
+            return;
+        }
 
-// resolve an entity
-const resolution = await client.resolution.resolution({name: "Victoria Beckham"});
+        // Fetch details of the best match
+        const entityDetails = await client.entity.getEntity(resolution.data[0].entityId);
```

Comment:
Happy to defer to views on this one.

I think this is more structured and more importantly this can be 'drag-dropped' into anyone wanting to use/extend the code. :-)

Reply:
Yeah, I don't feel strongly either. I tend to view hello world examples as being the minimal code needed to test an endpoint and not something people should consider a production ready example.... at the same time, it is easy enough to read. I will approve whatever is here when this PR is sent back for re-review :)